pkg/util/ftp/ftp.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2024 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ftp
+
+import (
+	"fmt"
+	"io"
+	"net/url"
+	"path"
+	"time"
+
+	"github.com/pkg/sftp"
+	"github.com/spf13/afero"
+	"golang.org/x/crypto/ssh"
+	"k8s.io/klog/v2"
+)
+
+const (
+	uploadToFileServerTries      = 5
+	uploadToFileServerRetryDelay = 5 * time.Second
+)
+
+func ParseFTPUploadUrl(uploadUrl string) (*url.URL, error) {
+	parsedURL, err := url.Parse(uploadUrl)
+	if err != nil {
+		return nil, err
+	}
+	if parsedURL.Scheme != "sftp" {
+		return nil, fmt.Errorf("not sftp protocol")
+	}
+	return parsedURL, nil
+}
+
+type Uploader interface {
+	// Upload uploads a file to the target sftp address using ssh config.
+	Upload(url string, fileName string, config *ssh.ClientConfig, outputFile afero.File) error
+}
+
+type SftpUploader struct {
```

Comment:
```suggestion
type SFTPUploader struct {
```
Please check other similar occurrences to replace ftp with FTP.